### PR TITLE
[Scripts] Adding Windows scripts

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR introduces a set of new build scripts for Windows OS. These scripts are added to the "scripts" directory and serve various purposes. Here's a brief overview of the changes made:

1. `scripts/build.bat`: This script sets up the compiler and various environment variables, then configures and builds the project using CMake. It also includes options to control the build type, enable/disable testing, and build components like C, Python, and Fortran.

2. `scripts/build_and_test.bat`: This script is designed to be run from the build directory. It invokes the `scripts/build.bat` script with specific options and then runs CTests for testing the project. Afterward, it runs Python tests.

3. `scripts/build_c.bat`: This script is used to build the C component of the project by calling the `scripts/build.bat` script with appropriate settings.

4. `scripts/build_cpp.bat`: This script is used to build the C++ component of the project by calling the `scripts/build.bat` script.

5. `scripts/build_python.bat`: This script is used to build the Python component of the project by calling the `scripts/build.bat` script.

In summary, this PR adds new build scripts to the project, making it easier to configure, build, and test different components of the project, such as C, C++, and Python, with various options and settings in the Windows operative system.

Fixes https://github.com/KratosMultiphysics/CoSimIO/issues/359 and https://github.com/KratosMultiphysics/CoSimIO/issues/360

### CODE REVIEW

File: scripts/build.bat [hunk 1]
```diff
@@ -0,0 +1,53 @@
+@echo off
+rem Please do not modify this script. Copyt into the build folder
```
Comment (philbucher) [inline scripts/build.bat:2]:
```suggestion
rem Please do not modify this script. Copy it into the build folder
```

File: scripts/build.bat [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+rem Set defaults
+if not defined CO_SIM_IO_BUILD_TESTING set CO_SIM_IO_BUILD_TESTING=ON
+if not defined CO_SIM_IO_BUILD_C set CO_SIM_IO_BUILD_C=OFF
+if not defined CO_SIM_IO_BUILD_PYTHON set CO_SIM_IO_BUILD_PYTHON=OFF
+if not defined CO_SIM_IO_BUILD_FORTRAN set CO_SIM_IO_BUILD_FORTRAN=OFF
+if not defined CO_SIM_IO_STRICT_COMPILER set CO_SIM_IO_STRICT_COMPILER=ON
+
+rem Warning: In windows this option only works if you run through a terminal with admin privileges
+rem set COSIMIO_INSTALL_PYTHON_USING_LINKS=ON
+
```
Comment (philbucher) [inline scripts/build.bat:23]:
```suggestion
```

this does not exist

File: scripts/build_python.bat [hunk 3]
```diff
@@ -0,0 +1,6 @@
+@echo off
+REM Should be copied to the build directory
```
Comment (philbucher) [inline scripts/build_python.bat:2]: why?

File: scripts/build.bat [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+if not defined NUMBER_OF_COMPILATION_CORES set NUMBER_OF_COMPILATION_CORES=%NUMBER_OF_PROCESSORS%
+if not defined COSIMIO_SOURCE set COSIMIO_SOURCE=%~dp0..
+if not defined COSIMIO_BUILD set COSIMIO_BUILD=%COSIMIO_SOURCE%/build
+
+rem Set defaults
+if not defined CO_SIM_IO_BUILD_TESTING set CO_SIM_IO_BUILD_TESTING=ON
+if not defined CO_SIM_IO_BUILD_C set CO_SIM_IO_BUILD_C=OFF
+if not defined CO_SIM_IO_BUILD_PYTHON set CO_SIM_IO_BUILD_PYTHON=OFF
+if not defined CO_SIM_IO_BUILD_FORTRAN set CO_SIM_IO_BUILD_FORTRAN=OFF
+if not defined CO_SIM_IO_STRICT_COMPILER set CO_SIM_IO_STRICT_COMPILER=ON
```
Comment (philbucher) [inline scripts/build.bat:19]: this seems superfluous, it is already set in cmake

**Review by philbucher — COMMENTED** [review]
some questions

Also, can you please incorporate the comment of Pooyan?

File: scripts/build_python.bat (hunk 3 above)
Comment (loumalouomega) [reply scripts/build_python.bat:2]: Because the source directory is estimated assuming it is copied into build directory

File: scripts/build.bat (hunk 4 above)
Comment (loumalouomega) [reply scripts/build.bat:19]: True, I added for debugging poruposes

**Comment by loumalouomega** [comment]
> some questions
> 
> Also, can you please incorporate the comment of Pooyan?

It is already added into the build.bat if you take a look

**Review by philbucher — APPROVED** [review]
thanks!

**Comment by philbucher** [comment]
last thing, could you please update the [docs](https://kratosmultiphysics.github.io/CoSimIO/tutorials/cpp/integration_co_sim_io.html) to point to the new files? Thx

**Comment by loumalouomega** [comment]
> last thing, could you please update the [docs](https://kratosmultiphysics.github.io/CoSimIO/tutorials/cpp/integration_co_sim_io.html) to point to the new files? Thx

Okay

**Comment by loumalouomega** [comment]
Done
